Mathlib/RingTheory/DualNumber.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2024 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Algebra.DualNumber
+import Mathlib.RingTheory.LocalRing.Defs
+import Mathlib.RingTheory.Nilpotent.Basic
+
+/-!
+# Algebraic properties of dual numbers
+
+## Main results
+
+* `DualNumber.instLocalRing`
+The dual numbers over a field `K` form a local ring.
+
+-/
+
+namespace DualNumber
+
+lemma isNilpotent_inr {R : Type*} [Semiring R] (r : R) :
+    IsNilpotent (.inr r : R[ε]) :=
+  ⟨2, by rw [pow_two, inr_eq_smul_eps]; ext <;> simp⟩
+
+lemma isNilpotent_eps {R : Type*} [Semiring R] :
+    IsNilpotent (ε : R[ε]) :=
+  isNilpotent_inr 1
+
+section Field
+
+variable {K : Type*} [Field K]
+
+-- TODO: delete when #12125 is merged
+lemma isUnit_inl {r : K} (hr : r ≠ 0) :
+    IsUnit (.inl r : K[ε]) :=
+  hr.isUnit.map (TrivSqZeroExt.inlAlgHom K K K)
+
+lemma isUnit_or_isNilpotent (a : K[ε]) : IsUnit a ∨ IsNilpotent a := by
+  by_cases ha : a.fst = 0
```

Comment:
Does casing on `IsUnit a ∨ IsNilpotent a` work here? That is, are dual numbers over local rings also local rings? Certainly the unit case is true, I haven't thought about the nilpotent I've.

Reply:
Not sure what you mean by casing on `IsUnit a ∨ IsNilpotent a` -- it's not the case that in a local ring, something is either unit or nilpotent. Counterexample: k[[X]], the ring of power series over a field, has unique maximal ideal (X) but that ideal isn't nilpotent or unit.